File: app/src/main/java/com/ne1c/developerstalk/ui/views/RoomsListView.java
package com.ne1c.developerstalk.ui.views;

import android.support.annotation.StringRes;

import com.ne1c.developerstalk.models.view.RoomViewModel;

import java.util.ArrayList;
import java.util.List;

public interface RoomsListView {
    void showRooms(List<RoomViewModel> rooms);

    void showError(@StringRes int resId);

    void showDialog();

    void dismissDialog();

    void errorSearch();

    void resultSearch(ArrayList<RoomViewModel> rooms);

    void resultSearchWithOffset(ArrayList<RoomViewModel> rooms);
}

File: app/src/main/kotlin/com/ne1c/gitteroid/ui/views/RoomsListView.kt
package com.ne1c.gitteroid.ui.views

import android.support.annotation.StringRes

import com.ne1c.gitteroid.models.view.RoomViewModel

import java.util.ArrayList

interface RoomsListView {
    fun showRooms(rooms: List<RoomViewModel>, fresh: Boolean)

    fun showError(@StringRes resId: Int)

    fun showDialog()

    fun dismissDialog()

    fun errorSearch()

    fun resultSearch(rooms: ArrayList<RoomViewModel>)

    fun resultSearchWithOffset(rooms: ArrayList<RoomViewModel>)
}
